nuget/Microsoft.Windows.CsWinRT.targets

```diff
@@ -147,6 +147,44 @@ $(CsWinRTFilters)
     <Copy SourceFiles="$(CsWinRTGeneratedFilesDir)\$(AssemblyName).winmd" DestinationFolder="$(TargetDir)" UseHardlinksIfPossible="false" SkipUnchangedFiles="true" />
   </Target>
 
+  <!-- ====================== -->
+  <!-- Call The GuidPatch tool on the projection .dll -->
+  <Target Name="CsWinRTInvokeGuidPatcher" Condition="'$(CsWinRTEnableGuidPatch)' == 'true'" AfterTargets="Compile" BeforeTargets="Link">
+    <!-- We need to invoke GuidPatch.exe. This lives in two different places depending on our environment. 
+
+         If we are in the CsWinRT repo, we don't have a .nupkg to use to find the exe, 
+           then we need to rely on CsWinRTGuidPatchRepoPath and CsWinRTGuidPatchRuntimePath from src/Directory.Build.props 
+           to get a hold of the files needed 
+           
+         If we are not in the CsWinRT repo, we can use the package paths we know -->
+
+    <PropertyGroup Condition="'$(CsWinRTInRepo)' == 'true'">
+      <!-- We use _DirectoryBuildTargetsBasePath as it is the only way I found to get "c:/.../cswinrt/src" -->
+      <CsWinRTGuidPatchExePath>$(_DirectoryBuildTargetsBasePath)\$(CsWinRTGuidPatchRepoPath)</CsWinRTGuidPatchExePath>
+      <CsWinRTGuidPatchRuntimeDll>$(_DirectoryBuildTargetsBasePath)\$(CsWinRTGuidPatchRuntimePath)</CsWinRTGuidPatchRuntimeDll> 
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(CsWinRTInRepo)' != 'true'">
+      <CsWinRTGuidPatchExePath>$(CsWinRTPath)build\tools\</CsWinRTGuidPatchExePath>
+      <CsWinRTGuidPatchRuntimeDll>$(CsWinRTPath)lib\net5.0\</CsWinRTGuidPatchRuntimeDll>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <CsWinRTGuidPatchInput Condition="'$(CsWinRTGuidPatchInput)' == ''">@(BuiltProjectOutputGroupKeyOutput->'%(Identity)')</CsWinRTGuidPatchInput>
+    </PropertyGroup>
+
+    <Exec Command="$(CsWinRTGuidPatchExePath)GuidPatch.exe $(CsWinRTGuidPatchInput) $(CsWinRTGuidPatchRuntimeDll)" ConsoleToMsBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="CsWinRTGuidPatchOutput" />
+    </Exec>
+    
+    <ItemGroup>
+      <PatchedFiles Include="$(_DirectoryBuildTargetsBasePath)\GuidPatcherOutput\*.dll" />
```

Comment:
How is the output folder decided on?

Reply:
https://github.com/microsoft/CsWinRT/blob/d6f5c04fe4ba0364e74b14b22be3cef35a71be78/src/Perf/GuidPatch/Program.cs#L31

Reply:
I meant the parent folder this is created in.

Reply:
The Exec task runs and it's "WorkingDirectory is the project's current working directory [(source)](https://docs.microsoft.com/en-us/visualstudio/msbuild/exec-task?view=vs-2019)

We could set that property when we call Exec to something, then use that property when we go to Copy